### ISSUE TITLE
Runtime-dependency on cabal executable?

### ISSUE THREAD

**Issue by nh2** [input-output-hk/stack2nix#100]
I got this:

```
Ensuring cabal version is >= 2 ...
stack2nix: ERROR: cabal version must be 2 or higher. Current version: 1.22.6.0
CallStack (from HasCallStack):
  error, called at src/Stack2nix/Util.hs:47:49 in stack2nix-0.1.3.1-3EqQ1tT1vMJGpAYSWfmhUQ:Stack2nix.Util
```

I see it calls `cabal --version`. Why is that?

`stack2nix` already link in the `Cabal` library, so couldn't it use only functionality from that? Then it wouldn't care about whatever `cabal` is in path (or even require it to be installed).

**Comment by nh2** [comment]
Another question: For now, how can I get exactly a version of the `cabal` executable that's known to work with `stack2nix`? Can I obtain it as part of `stack2nix`'s nix-shell environment?

**Comment by domenkozar** [comment]
`cabal update` is still ran to make sure hackage db is up to date. It could be made optional.

iirc stack2nix did provision itself a cabal from nixpkgs, are you using master branch?

**Comment by nh2** [comment]
@domenkozar Yeah I'm in `9070f9173ae32f0be6f7830c41c8cfb8e780fdbf`. In `nix-shell --pure default.nix` there's no `cabal` in PATH. Am I looking in the wrong spot?

**Comment by domenkozar** [comment]
Relevant #102
